### PR TITLE
Intelligently adjust field for newly added action

### PR DESCRIPTION
**Pull request by j-f1**
✨

https://user-images.githubusercontent.com/25517624/220503404-2ddb046c-fc2f-4c19-a825-1a6a7104e84a.mov

### CODE REVIEW

File: packages/desktop-client/src/components/modals/EditRule.js [hunk 1]
```diff
@@ -397,13 +396,18 @@ function newInput(item) {
...
   conditions,
-  conditionFields,
   editorStyle,
   isSchedule,
   onChangeConditions,
 }) {
   function addCondition(index) {
-    let field = 'payee';
+    // (remove the inflow and outflow pseudo-fields since they’d be a pain to get right)
+    let fields = conditionFields.slice(0, -2).map(f => f[0]);
```
Comment (MatissJanis) [inline packages/desktop-client/src/components/modals/EditRule.js:405]: **💬 suggestion:** ‏any other way how we could remove these specific fields?

I'm afraid that currently if someone decides to re-order the values in `conditionFields` we'll end up with a bug that's very hard to spot.

File: packages/desktop-client/src/components/modals/EditRule.js [hunk 2]
```diff
@@ -397,13 +396,18 @@ function newInput(item) {
...
+    for (let cond of conditions) {
+      fields = fields.filter(f => f !== cond.field);
+    }
```
Comment (MatissJanis) [inline packages/desktop-client/src/components/modals/EditRule.js:408]: **🥜 nitpick:** ‏absolute nitpick as the data size is very tiny, so feel free to ignore

Current time complexity is O(n^2). If you changed the lookup to a hashmap, you could get this down to O(n).

File: packages/desktop-client/src/components/modals/EditRule.js [hunk 3]
```diff
@@ -617,7 +620,11 @@ export default function EditRule({
   }
 
   function addAction(index) {
-    let field = 'category';
+    let fields = actionFields.map(f => f[0]);
+    for (let action of actions) {
+      fields = fields.filter(f => f !== action.field);
+    }
```
Comment (MatissJanis) [inline packages/desktop-client/src/components/modals/EditRule.js:626]: same

File: packages/loot-core/src/shared/rules.js [hunk 4]
```diff
@@ -57,8 +57,6 @@ export function mapField(field, opts) {
       return 'amount (inflow)';
     case 'amount-outflow':
       return 'amount (outflow)';
-    case 'cleared':
-      return 'cleared';
```
Comment (MatissJanis) [inline packages/loot-core/src/shared/rules.js:61]: **❓ question:** ‏Why was this removed?

Comment (MatissJanis) [reply packages/loot-core/src/shared/rules.js:61]: oh, this just makes the name "pretty". and since var name matches what we return - there's no point in this. :)

File: packages/desktop-client/src/components/modals/EditRule.js (hunk 2 above)
Comment (j-f1) [reply packages/desktop-client/src/components/modals/EditRule.js:408]: I did it this way intentionally so that the fields are added in a defined priority order, which I don’t think you can do with a hash map?

Comment (MatissJanis) [reply packages/desktop-client/src/components/modals/EditRule.js:408]: Something like this (haven't tested):

```
const hashmap = new Set(conditions.map(({ field }) => field));
const newFields = fields.filter(f => !hashmap.has(f));
```

**Review by MatissJanis — APPROVED** [review]
